gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala

```diff
@@ -364,6 +364,18 @@ case class GremlinScala[End, Labels <: HList](traversal: GraphTraversal[_, End])
     GremlinScala[A, Labels](traversal.union(mappedTraversals: _*))
   }
 
+  def choose[A, B](
+    predicate: GremlinScala[End, HNil] ⇒ GremlinScala[A, _],
+    trueChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _],
+    falseChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _]): GremlinScala[B, Labels] = {
+    val p: Traversal[_, A] = predicate(start).traversal
```

Comment:
I'm typically leaning towards more type level consistency, and this is a place where I would definitely go for Boolean. Making empty traversal equal true doesn't make sense to me, I'd rather require the user to make a clear statement. And if they care about empty traversals, they can use `exists`